### DISCUSSION TITLE
How do I retrieve the export of the latest version of a pub through the API?

### DISCUSSION THREAD

**Discussion by lukvmil** [knowledgefutures/pubpub#3028] [General]
Not sure if this is a bug, but using the /api/export endpoint, historyKey does not default to the latest version like it says, but instead returns the first version which is a blank document. Manually setting to -1 also does not work, and the only solution I have right now is setting it to an arbitrary large number which doesn't seem ideal. The API also doesn't seem to return the latest historyKey anywhere else. The documentation says it's supposed to be return from the /api/pubs endpoint, but all of the pubs are missing this key when I make this call. Help would be appreciated!

**Comment by lukvmil** [comment]
Update: I was able to find the latest historyKey through the /api/pubHistory endpoint, but this doesn't seem to be documented at all. It's not listed in the API Docs and I was only able to find it by poking through the source code and seeing how the front end knows which version to export. If there's a better way to do this please let me know, but it seems like there's some discrepancies between the codebase and documentation right now.

**Reply by tefkah** [reply]
Hi, thanks for reporting this! That does indeed seem to be a bug, ill have a look

**Reply by luluboxproapk** [reply]
To retrieve the latest version export of a pub through the API, make sure you're calling the correct endpoint and checking version parameters properly. Just like tools such as lulubox pro apk focus on version compatibility, API integration also requires accurate version control. also visit this site for better experience. https://theluluboxpro.com/

**Comment by saqi004** [comment]
It sounds like you’re encountering an issue with the /api/export endpoint not defaulting to the latest historyKey. This could be a bug. Here are a few steps to troubleshoot:

Check API Documentation: Make sure you're following the latest version of the API documentation for any updates or changes.
Review API Calls: Double-[check the](https://menuland.ph/subway-menu-in-philippines/) request parameters and ensure there are no errors in how you’re calling the endpoint.
Contact Support: If the issue persists, consider reaching out to the API support team with detailed information about your problem, including the endpoint, parameters used, and any response you received.
Using an arbitrary large number for historyKey isn’t ideal, and you shouldn’t have to do that. Hopefully, the support team can provide clarity on this.

**Comment by careerpro14** [comment]
Did any solution work? @lukvmil
